specifyweb/frontend/js_src/lib/components/QueryBuilder/Components.tsx

```diff
@@ -173,7 +173,10 @@ export function QueryButton({
       dialogMessage={queryText.queryDeleteIncompleteDescription()}
       disabled={disabled}
       showConfirmation={showConfirmation}
-      onConfirm={handleClick}
+      onConfirm={(): void => {
+        handleClick();
+        globalThis.location.reload();
```

Comment:
the component you are modifying (QueryButton) is used in a lot of places - you certainly don't want to cause page reload in all those places.
please remember to check for usages when modifying some component so that you can see all the places it could be affecting!